src/gui/MainWindow.cpp

```diff
@@ -519,6 +519,11 @@ void MainWindow::updateWindowTitle()
         windowTitle = BaseWindowTitle;
     } else {
         windowTitle = QString("%1 - %2").arg(customWindowTitlePart, BaseWindowTitle);
+        if (customWindowTitlePart.right(1) == "*") {
```

Comment:
This would not work if the database that is modified is not currently selected. It also perpetuates the terrible behavior of checking the title for database modifications (not your fault).